src/components/checkbox/checkbox.test.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { createRef } from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Checkbox } from './checkbox'
+
+test('renders checkbox label correctly', () => {
+  render(
+    <Checkbox
+      label="Checkbox label"
+      onChange={(event) => {
+        event?.target.checked
+      }}
+    />,
+  )
+  expect(screen.getByText('Checkbox label')).toBeInTheDocument()
+})
+
+test('onChange event of Checkbox working correctly', () => {
+  const onChange = jest.fn((event) => event.target.checked)
+  render(<Checkbox label="Checkbox label" onChange={onChange} />)
+  fireEvent.click(screen.getByText('Checkbox label'))
+  expect(onChange).toBeCalled()
+  expect(onChange.mock.results[0].value).toBe(true)
+})
```

Comment:
We also need a test to check that the `onChange` works correctly when you are toggling from **indeterminate** state.